build/purger/purger.py

```diff
@@ -229,6 +277,199 @@ def wait_and_purge_room(room_id: str = None, event_id: str = None) -> None:
+    Each server is responsible for its own users
+
+    :param api: Api object to own server
+    :param global_user_activity: content of user activity file
+    :param network_to_broadcast_rooms: RoomInfo Objects of all broadcast rooms
+    :return: None
+    """
+
+    # perform update on user presence for due users
+    # receive a list for due users on each network
+    due_users = update_user_activity(api, global_user_activity, network_to_broadcast_rooms)
+    # purge due users form rooms
+    purge_inactive_users(api, global_user_activity, network_to_broadcast_rooms, due_users)
+
+    return global_user_activity
+
+
+def get_network_to_broadcast_rooms(api: GMatrixHttpApi) -> Dict[str, List[RoomInfo]]:
+
+    room_alias_fragments = [
+        DISCOVERY_DEFAULT_ROOM,
+        PATH_FINDING_BROADCASTING_ROOM,
+        MONITORING_BROADCASTING_ROOM,
+    ]
+
+    server = urlparse(api.base_url).netloc
+    network_to_broadcast: Dict[str, List[RoomInfo]] = dict()
+
+    for network in Networks:
+        broadcast_room_infos: List[RoomInfo] = list()
+        network_to_broadcast[str(network.value)] = broadcast_room_infos
+        for room_alias_fragment in room_alias_fragments:
+            broadcast_room_alias = make_room_alias(ChainID(network.value), room_alias_fragment)
+            local_room_alias = f"#{broadcast_room_alias}:{server}"
+
+            try:
+                room_id = api.get_room_id(local_room_alias)
+                broadcast_room_infos.append(RoomInfo(room_id, broadcast_room_alias, server))
+            except MatrixError as ex:
+                click.secho(f"Could not find room {broadcast_room_alias} with error {ex}")
+
+    return network_to_broadcast
+
+
+def update_user_activity(
+    api: GMatrixHttpApi,
+    global_user_activity: Dict[str, Any],
+    broadcast_rooms: Dict[str, List[RoomInfo]],
+) -> Dict[str, List[str]]:
+    """
+    runs update on users' presences which are about to be kicked.
+    If they are still overdue they are going to be added to the list
+    of users to be kicked. Presence updates are included in the
+    global list which is going to be stored in the user_activity_file
+    :param api: api to its own server
+    :param global_user_activity: content of user_activity_file
+    :param broadcast_rooms: room information for broadcast rooms
+    :return: a list of due users to be kicked
+    """
+    current_time = int(time.time())
+    last_user_activity_update = global_user_activity["last_update"]
+    network_to_users = global_user_activity["network_to_users"]
+    network_to_due_users = dict()
+    fetch_new_members = False
+
+    # check if new members have to be fetched
+    # new members only have to be fetched every
+    # USER_PURGING_THRESHOLD since that is the earliest time
+    # new members are able to be kicked. This keeps the load on
+    # the server as low as possible
+    if last_user_activity_update < current_time - USER_PURGING_THRESHOLD:
```

Comment:
This means, that on first run, without https://github.com/raiden-network/raiden-service-bundle/pull/89/files#diff-70a9de6e8cf31fa11e7531710c0a2377R211 `USER_ACTIVITY_PATH` not existing,  we will skip everything, right?

Reply:
nice nitpick but it would be nicer to start right away :) Im gonna change it